docs/developers/smart-contract-tutorial-foundry.md

```diff
@@ -0,0 +1,208 @@
+        counter.incrementCounter();
+        counter.incrementCounter();
+        int value = counter.getCount();
+        assertEq(value, 12);
+        emit log_int(value);
+    }
+
+    function testDecrement() public {
+        counter.decrementCounter();
+        int value = counter.getCount();
+        assertEq(value, 9);
+        emit log_int(value);
+    }
+}
+```
+
+Foundry uses [Dappsys Test](https://book.getfoundry.sh/reference/ds-test.html) to provide basic logging and assertion functionality. It's included in the Forge Standard Library.
+
+Here, we are using `assertEq` to assert equality. You can view all of the assertion functions available [here](https://book.getfoundry.sh/reference/ds-test.html?highlight=log_int#asserting).
+
+Next, we can test the contract using __Forge__ with the following command:
+
+```sh
+forge test -vv
+```
+
+### Updating the deployment script
+
+Now that we've tested the contract, let's try deploying it locally using [Solidity Scripting](https://book.getfoundry.sh/tutorials/solidity-scripting.html).
+
+To do so, update the deloyment script at `script/Contracts.s.sol` with the following code:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "forge-std/Script.sol";
+
+import {Counter} from "src/Counter.sol";
+
+contract ContractScript is Script {
+    function setUp() public {}
+
+    function run() public {
+        vm.startBroadcast();
+        new Counter(10);
+        vm.stopBroadcast();
+    }
+}
+```
+
+Now we can use this script to deploy our smart contract to either a live or test network.
+
+### Deploying locally
+
+Next start Anvil, the local testnet:
+
+```sh
+anvil
+```
+
+Once started, Anvil will give you a local RPC endpoint as well as a handful of Private Keys and Accounts that you can use.
+
+We can now use the local RPC along with one of the private keys to deploy locally:
+
+```sh
+forge script script/Contract.s.sol:ContractScript --fork-url http://localhost:8545 \
+--private-key $PRIVATE_KEY --broadcast
+```
+
+Once the contract has been deployed locally, we can test sending transactions to it with `cast send`:
+
+```sh
+cast send <contract-address> "incrementCounter()" \
+--private-key $PRIVATE_KEY
+```
+
+We can then perform read operations with `cast call`:
+
+```sh
+cast call <contract-address> "getCount()(int)"
+```
+
+### Deploying to Celestia Ethermint
+
+Now that we've deployed and tested locally, we can deploy to Celestia Ethermint.
+
+> Be sure to be using the private key for the account which holds the test CET tokens requested from the faucet.
+
+To do so, run the following script:
+
+```sh
+forge script script/Contract.s.sol:ContractScript --rpc-url http://35.208.160.145:8545 \
+ --private-key $PRIVATE_KEY --broadcast
+```
+
+Once the contract has been deployed to Celestia Ethermint, we can use `cast send` to test sending transactions to it:
+
+```sh
+cast send <contract-address> "incrementCounter()" --rpc-url http://35.208.160.145:8545 \
```

Comment:
Also here, we can tell the developer to set the env var CONTRACT_ADDRESS based on the response from `forge script`
```sh
CONTRACT_ADDRESS=<value>
```
then do 
```sh
cast send $CONTRACT_ADDRESS ...
```
so it's easy for them to copy paste after setting the env var